### PR TITLE
Feat/profile

### PR DESCRIPTION
**Pull request by ninadarbaidze**

### CODE REVIEW

File: components/Profile/EditProfile/types.d.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export type ProfileInfoTypes = {
+  userData: any;
```
Comment (GigaGiorgadze) [inline components/Profile/EditProfile/types.d.ts]: no any >:(

Comment (GigaGiorgadze) [reply components/Profile/EditProfile/types.d.ts]: do u have good reason for any?

File: components/Profile/ProfileForm/ProfileForm.tsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+import { useProfileForm } from './useProfileForm';
+import { UpdatePassTypes } from './types';
+import { Input, Button } from 'components';
+
+const ProfileForm: React.FC<UpdatePassTypes> = (props) => {
+  const { updatePassword, setUpdatePassword, imageChangeHandler } = props;
+  const { t, formik } = useProfileForm();
+  const fileRef = useRef<HTMLInputElement>(null);
+
+  const changeHandler = (e: any) => {
```
Comment (GigaGiorgadze) [inline components/Profile/ProfileForm/ProfileForm.tsx]: no any this is ChangeEvent<HTMLInputElement>

File: components/Profile/ProfileForm/ProfileForm.tsx [hunk 3]
```diff
@@ -0,0 +1,110 @@
+import { useRef } from 'react';
+import { useProfileForm } from './useProfileForm';
+import { UpdatePassTypes } from './types';
+import { Input, Button } from 'components';
+
+const ProfileForm: React.FC<UpdatePassTypes> = (props) => {
+  const { updatePassword, setUpdatePassword, imageChangeHandler } = props;
+  const { t, formik } = useProfileForm();
+  const fileRef = useRef<HTMLInputElement>(null);
```
Comment (GigaGiorgadze) [inline components/Profile/ProfileForm/ProfileForm.tsx]: move this stuff inside custom hook, not just here but everywhere, this file should have only custom hook and jsx nothing else

File: components/Profile/ProfileForm/useProfileForm.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+import { useSession } from 'next-auth/react';
+
+export const useProfileForm = () => {
+  const { t } = useTranslation();
+  const ctx = useContext(AuthContext);
+  const userCtx = useContext(UserContext);
+  const { data: session } = useSession();
+  const router = useRouter();
+
+  const onSubmit = async (values: any) => {
```
Comment (GigaGiorgadze) [inline components/Profile/ProfileForm/useProfileForm.ts]: no any

File: components/Profile/ProfileForm/useProfileForm.ts [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    const userId: any = session ? session.userId : ctx.userId;
+    const token: any = session ? session.accessToken : ctx.token;
+    const formData = new FormData();
+    formData.append('image', values.image);
+    formData.append('username', values.username);
+    formData.append('email', values.email);
+    formData.append('password', values.password);
```
Comment (GigaGiorgadze) [inline components/Profile/ProfileForm/useProfileForm.ts]: why not just loop over vlues

File: components/Profile/SideBar/SideBar.tsx [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+  const imagePreviewHandler = () => {
+    const defaultProfileImg = `/assets/images/profile.png`;
+
+    if (!userCtx.userState.profileImage && session?.user) {
+      return defaultProfileImg;
+    } else if (session?.user) {
+      return session!.user.image as any;
+    } else {
+      return `${process.env.NEXT_PUBLIC_API_URL}/${userCtx.userState.profileImage}`;
```
Comment (GigaGiorgadze) [inline components/Profile/SideBar/SideBar.tsx]: also move this inside custom hook, and as i said before everywhere else,

File: pages/feed/profile/index.tsx [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+
+const Profile = () => {
+  const [mobMenu, setMobMenu] = useState(false);
+  const router = useRouter();
+  const ctx = useContext(AuthContext);
+  const userCtx = useContext(UserContext);
+  const { data: session, status } = useSession();
+
+  useEffect(() => {
+    console.log(ctx.isLoggedIn);
```
Comment (GigaGiorgadze) [inline pages/feed/profile/index.tsx]: remove console log

File: services/profileRequests.ts [hunk 8]
```diff
@@ -0,0 +1,22 @@
+import axios from './axios';
+import { AxiosResponse } from 'axios';
+
+export const updateProfile = async (
+  data: FormData,
+  token: string
+): Promise<AxiosResponse<any>> => {
```
Comment (GigaGiorgadze) [inline services/profileRequests.ts]: give this return type, first can be any cuz its error if i remember but second parameter should be return type

File: store/AuthContext.tsx [hunk 9]
```diff
@@ -1,23 +1,64 @@
...
   const [loginModalState, setloginModalState] = useState(false);
   const [passwordRecoveryState, setPasswordRecoveryState] = useState(false);
   const [passwordUpdateState, setPasswordUpdateState] = useState(false);
 
+  let initialToken;
+  if (storedData) {
+    initialToken = storedData.storedToken;
+  }
+
+  const [token, setToken] = useState(initialToken) as any;
```
Comment (GigaGiorgadze) [inline store/AuthContext.tsx]: no any

File: store/UserContext.tsx [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+export const UserContext = createContext({
+  userState: initialState,
+  getUser: (data: any) => {},
+});
+
+export const UserContextProvider: React.FC<Children> = (props) => {
+  const [userState, dispatchUserAction] = useReducer(reducer, initialState);
+
+  const getUser = (data: any) => {
+    dispatchUserAction({
```
Comment (GigaGiorgadze) [inline store/UserContext.tsx:28]: why u need this context? don't u have next auth useSession?

**Review by GigaGiorgadze — CHANGES_REQUESTED** [review]

File: store/UserContext.tsx (hunk 10 above)
Comment (ninadarbaidze) [reply store/UserContext.tsx:28]: I'm storing user-related data here, planning to add movies related context and that's it. can I keep it here? authcontext is leaner this way

Comment (GigaGiorgadze) [reply store/UserContext.tsx:28]: may i ask why u need authContext?

File: components/Profile/SideBar/helper.ts [hunk 11]
```diff
@@ -0,0 +1,14 @@
+export const imagePreviewHandler = (
+  userCtx: { userState: { profileImage: string } },
+  session: any
```
Comment (GigaGiorgadze) [inline components/Profile/SideBar/helper.ts]: why is sessionAny? is this next-auth session? if so I think that has type Session

**Review by GigaGiorgadze — CHANGES_REQUESTED** [review]

**Review by GigaGiorgadze — APPROVED** [review]

**Pull request merged**